services/storage/src/simcore_service_storage/dsm.py

```diff
@@ -961,17 +962,27 @@ async def delete_project_simcore_s3(
 
         async with self.engine.acquire() as conn, conn.begin():
             # access layer
-            can: Optional[AccessRights] = await get_project_access_rights(
-                conn, int(user_id), project_id
-            )
-            if not can.delete:
-                logger.debug(
-                    "User %s was not allowed to delete project %s",
-                    user_id,
-                    project_id,
+            try:
+                can: Optional[AccessRights] = await get_project_access_rights(
+                    conn, int(user_id), project_id
                 )
-                raise web.HTTPForbidden(
-                    reason=f"User does not have delete access for {project_id}"
+                if not can.delete:
```

Comment:
@pcrespov I don't really see how delayed deletion would change anything at this point. this just would *delay* the issue. 
But I agree that for delete/undelete we could do that.

Reply:
@sanderegg  Here is my reasoning: If the deletion is delayed, the project will get "marked" (not deleted) and we can still recover the access rights (because the info still there) when we are about to "mark" (not delete) the data.

Then the actual deletion is delegated to another background task/service/process ...  (that is something else but happens much later)

Reply:
It can be slightly better, but you would have to invert the order fo deletion: first S3 content, then the project entry in the database, this will work with the marked for delete.

Reply:
@pcrespov yes you can still recover the project, if you first delete the S3 storage, then you do not have the problem. But that is delayed or not delayed. that makes no difference. 
What I say, is that in any case, you might have to handle that issue.